### PR TITLE
Fireworks 9_4_X: Fix bug in scaling after projected view destruction

### PR DESCRIPTION
**Pull request by alja**
#### PR description:
This is back-port of  #34655.

### CODE REVIEW

**Comment by alja** [comment]
+1

**Comment by perrotta** [comment]
@alja 
This backport is much more heavy than the PR in the master, or its 10_3 counterpart.
Please either restrict to backport only the real fix, or explain in the PR description why do you need all the rest.

**Comment by alja** [comment]
@perrotta 
The bugfix is isolated on 4 files that are within Fireworks/Calo event display. I don't know which part of this PR is concerning.

One of the event display users found a bug recently and this PR fixes. There is a workaround for it and the bug is not critical. I have tested it and I will send the test to the user as well today.

File: Fireworks/Calo/src/FWTauProxyBuilderBase.cc [hunk 1]
```diff
@@ -35,166 +36,140 @@
...
-         marker->AddLine(0., (phi>0 ? r*fabs(sin(theta)) : -r*fabs(sin(theta))), r*cos(theta),
-                         0., (phi>0 ? (r+size)*fabs(sin(theta)) : -(r+size)*fabs(sin(theta))), (r+size)*cos(theta) );
 
-      }
-      else
-      {
-         std::pair<double,double> phiRange = fireworks::getPhiRange( m_phis, phi );
-         double min_phi = phiRange.first-M_PI/36/2;
-         double max_phi = phiRange.second+M_PI/36/2;
-         if( fabs(phiRange.first-phiRange.first)<1e-3 ) {
```
Comment (perrotta) [inline Fireworks/Calo/src/FWTauProxyBuilderBase.cc:98]: This is one of the changes that were implemented in the master before #34655: is it still acceptable for 9_4_X?

**Comment by perrotta** [comment]
> @perrotta
> The bugfix is isolated on 4 files that are within Fireworks/Calo event display. I don't know which part of this PR is concerning.
> 
> One of the event display users found a bug recently and this PR fixes. There is a workaround for it and the bug is not critical. I have tested it and I will send the test to the user as well today.

@alja you copied the four files from the master to 9_4_X: that imports back in 9_4_X all other changes that were integrated in the meanwhile, not only this fix.

After a visula inspection, I think that all those extra modificiations are only the code-checks/code-formats updates that were integrated during the cleaning campaigns: however, I also noticed (at least) [another update](https://github.com/cms-sw/cmssw/pull/34680#pullrequestreview-718789865) that looks more relevant: perhaps is fine to have it also in 9_4_X, but it should be certified by you.

Even though a minimal backport with only the fixes would have been easier to evaluate and merge in a closed release, I think it can be still acceptable to merge this one in 9_4_X provided you verify and confirm that all the extra changes implemented (the one that I spotted and the possible other ones) are justified for that past release,

**Comment by alja** [comment]
@perrotta 
 Thank you for the through review. The check for minimum phi range was not existing in newer releases. I have inserted back the referenced segment back. This check make a segment marker in RhoPhi projection visible if there is a tiny phi range. I will keep an eye on this in the newer releases.

**Comment by alja** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**